src/NHibernate.Test/TestDialect.cs

```diff
@@ -120,10 +120,15 @@ public bool SupportsSqlType(SqlType sqlType)
 		/// </summary>
 		public virtual bool SupportsModuloOnDecimal => true;
 
+		/// <summary>
+		/// Supports sub-selects in order by clause
+		/// </summary>
+		public virtual bool SupportsSubSelectsInOrderBy => _dialect.SupportsScalarSubSelects;
```

Comment:
DB2 supports scalar sub selects, but it does not support them in order-by clause.

Reply:
And in SELECT statement seems too

Reply:
@bahusoid this test works just fine:

https://github.com/nhibernate/nhibernate-core/blob/6576a9675cbfc48b560c58f85794013b4512131d/src/NHibernate.Test/Linq/SelectionTests.cs#L408-L464